applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp

```diff
@@ -452,11 +453,11 @@ class ExplicitCentralDifferencesScheme
 
         // Solution of the explicit equation:
         if (nodal_mass > numerical_limit)
-            r_current_acceleration = r_current_residual / nodal_mass;
+            noalias(r_current_acceleration) = (r_current_residual - nodal_displacement_damping * r_current_velocity) / nodal_mass;
         else
-            r_current_acceleration = ZeroVector(3);
+            noalias(r_current_acceleration) = ZeroVector(3);
 
-        bool fix_displacements[3] = {false, false, false};
+        std::array<bool, 3> fix_displacements = {false, false, false};
 
         fix_displacements[0] = ((itCurrentNode)->pGetDof(DISPLACEMENT_X))->IsFixed();
```

Comment:
as a performance improvement i would use the position suggestion as it is done here. 

https://github.com/KratosMultiphysics/Kratos/blob/master/applications/FluidDynamicsApplication/custom_elements/fluid_element.cpp#L286

and 

https://github.com/KratosMultiphysics/Kratos/blob/master/applications/FluidDynamicsApplication/custom_elements/fluid_element.cpp#L291

this will (greatly) speedup the getdof search

Reply:
Ok